### DISCUSSION TITLE
How to set field?

### DISCUSSION THREAD

**Discussion by ghost** [airjp73/rvf#44] [Q&A] [answered]
Some component like DropToUpload, i want to set form field on the drop event, how to do that? 

In `react-hook-form`, there is a method `setValue` in `useForm`, does remix-validated-form have a similar method?

**Comment by airjp73** [comment, marked as answer]
Edit: As of `remix-validated-form` 4.3.0, I think [this](https://github.com/airjp73/remix-validated-form/discussions/44#discussioncomment-2536928) is the actual answer to this question.

We don't deviate from `remix` in that we don't keep track of the values at all and leave that up to the native HTML form. If you're integrating with a custom component the solution is usually to use a [hidden input](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/hidden) -- just like using remix without this library.

```tsx
const [value, setValue] = useState('');

return (
  <>
    <input type="hidden" name="myField" value={value} />
    <CustomComponent onChange={setValue} value={value} />
  </>
);
```

For file uploads, the might be slightly different. I'd recommend bringing this question up in a general remix place like the remix discord server or [discussions in the remix repo](https://github.com/remix-run/remix/discussions).

**Reply by alexinfurs** [reply]
Maybe having something like `setFieldValue`/`setFieldsValue` can be really useful without deviating too much from `remix` if, internally, `remix-validated-form` handles the `value` prop like you did in the example.
I've found myself facing a very complex forms and sometimes when a field changes, one o more fields need to be updated  according to some rules and it'd be useful if we can do something like this

```jsx
<>
  <input name="field1" />
  <input name="field2" onChange={(e) => e.target.value === "test" && setFieldValue('field1', "ok")} />
  <input name="field3" onChange={(e) => e.target.value === "test" && setFieldsValue({ field1: "ok", field2: "ok" })} />
</>
```
Do you suggest, eventually, another solution?

Also, if we keep track of values, we can conditionally render fields

```jsx
<>
  <input name="field1" />
  <input name="field2" disabled={fieldValues.field1 === "ok"} />
</>
```

**Reply by airjp73** [reply]
Hmm, yeah. I hadn't really considered the angle of updating _other_ fields. `react-hook-form` is able to make it work with uncontrolled inputs so I don't see why this library can't. I'll put some thought into this.

**Comment by airjp73** [comment]
`remix-validated-form` 4.3.0 now contains a `useControlField` hook that can be used to track the value of a field and modify it from anywhere.

**Reply by alexinfurs** [reply]
That's amazing! I can now switch from react-hook-form. Thanks a lot
